edx_name_affirmation/signals.py

```diff
@@ -0,0 +1,179 @@
+        is_onboarding(boolean): if the exam attempt is for an onboarding exam
+        is_proctored(boolean): if the exam attempt is for a proctored exam
+        backend_supports_onboarding(boolean): if the exam attempt is for an exam with a backend that supports onboarding
+    """
+    # We only care about updates from onboarding exams, or from proctored exams with a backend that does  not
+    # support onboarding. This is because those two event types are guaranteed to contain verification events,
+    # whereas timed exams and proctored exams with a backend that does support onboarding are not guaranteed
+    # to contain verification events.
+    if is_onboarding or (is_proctored and not backend_supports_onboarding):
+        translated_status = VerifiedNameStatus.translate_proctoring_status(status)
+        log_str = ''
+
+        verified_name = VerifiedName.objects.filter(
+            user__id=user_id,
+            proctored_exam_attempt_id=attempt_id
+        ).order_by('-created').first()
+        if verified_name:
+            # if a verified name for the given attempt ID exists, update it if the status should trigger a transition
+            if translated_status:
+                verified_name.status = translated_status
+                verified_name.save()
+                log_str = 'Updated VerifiedName for user={user_id} with proctored_exam_attempt_id={attempt_id} ' \
+                          'to have status={status} '
+        else:
+            # check if approved VerifiedName already exists for the user
+            verified_name = VerifiedName.objects.filter(
+                user__id=user_id,
+                status=VerifiedNameStatus.APPROVED
+            ).first()
+            if not verified_name:
+                if full_name and profile_name:
+                    # if they do not already have an approved VerifiedName, create one
+                    user = User.objects.get(id=user_id)
+                    verified_name = VerifiedName.objects.create(
+                        user=user,
+                        verified_name=full_name,
+                        proctored_exam_attempt_id=attempt_id,
+                        status=(translated_status if translated_status else VerifiedNameStatus.PENDING),
+                        profile_name=profile_name
+                    )
+                    log_str = 'Created VerifiedName for user={user_id} to have status={status} ' \
+                              'and proctored_exam_attempt_id={attempt_id}'
+                else:
+                    log.error(
+                        'Cannot create VerifiedName for user={user_id} for proctored_exam_attempt_id={attempt_id} '
+                        'because neither profile name nor full name were provided'.format(
+                            user_id=user_id,
+                            attempt_id=attempt_id,
+                        )
+                    )
+
+        if log_str:
```

Comment:
the late log makes the function less readable because the reader has to figure out that this if clause is just logging rather than logic. I think just embed the logging where it actually happens